### PR TITLE
[113] View/Edit profile (Frontend)

### PR DESCRIPTION
**Pull request by SaidganiUsm**
### Where

[View/Edit profile (Frontend)](https://jjbcapital.atlassian.net/browse/NL-113)

### What

Developed front-end for updating user profile changing password and user  profile itself

### How

Implemented all features for adding and deleting user photos, changing user data, and changing password

### Affected Area

Auctionify.API.ClientApp
Auctionify.Infrastructure

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

- [x] All Tests Passed, 0 Ignored, 0 Failed

### CODE REVIEW

**Review by otash8110 — CHANGES_REQUESTED** [review]
image
When I try to change user data I get validation error without message

image
- Log out is not working because of the wrong type of the dialog (Should be just Dialog) and overload method of the open. See other components which have such functionality.

image
- When I click on eye to show hidden password the form is submited. Also add some space between inputs because message cannot fit

**Review by khasanrashidov — APPROVED** [review]

**Review by khasanrashidov — DISMISSED** [review]

File: src/Auctionify.API/ClientApp/src/app/components/general/profile/update-user-profile/update-user-profile.component.scss [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        flex-direction: column;
+        gap: 16px;
+    }
+
+    &__about-me {
+        display: flex;
+        flex-direction: column;
+        gap: 24px;
+        min-width: 500px;
+        width: 1056px;
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/general/profile/update-user-profile/update-user-profile.component.scss:102]: Avoid specifying a specific element's width.
image

File: src/Auctionify.API/ClientApp/src/app/layout/navbar-seller/navbar-seller.component.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
                         <mat-icon>local_mall</mat-icon>
                         Active lots
                     </button>
-                    <button mat-flat-button color="white">
+                    <button mat-flat-button [routerLink]="['/profile']" color="white">
                         <mat-icon>account_circle</mat-icon>
                         MyProfile</button>
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/layout/navbar-seller/navbar-seller.component.html:33]: Add space between words

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
Fix the access to the profile page. It is currently available to an anonymous user

File: src/Auctionify.API/ClientApp/src/app/components/general/profile/update-user-profile/update-user-profile.component.html [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                        <mat-error *ngIf="profileForm.controls.aboutMe.hasError('invalidCharacters')">
+                            Only letters are allowed
+                        </mat-error>
+                        <mat-error *ngIf="profileForm.controls.aboutMe.hasError('minlength')">
+                            Minimum 30 characters required
+                        </mat-error>
+                    </mat-form-field>
+                </div>
+                <div class="main__personal-info">
+                    <h2>Presonal info</h2>
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/general/profile/update-user-profile/update-user-profile.component.html:114]: Fix typo

File: src/Auctionify.API/ClientApp/src/app/components/general/profile/user-profile/user-profile.component.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                <div class="main__name-date">
+                    <h1>{{ userProfileData?.firstName }} {{ userProfileData?.lastName }}</h1>
+                </div>
+            </div>
+            <div class="main__about-me">
+                <h2>About me</h2>
+                <p>{{ userProfileData?.aboutMe}}</p>
+            </div>
+            <div class="main__personal-info">
+                <h2>Presonal info</h2>
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/general/profile/user-profile/user-profile.component.html:45]: Fix typo

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
Profile page still available for an anonymous user

**Review by otash8110 — APPROVED** [review]

**Review by khasanrashidov — APPROVED** [review]

**Review by TsvirkoNikita — APPROVED** [review]

**Pull request merged**